scripts/check-gmp.gmp

```diff
@@ -0,0 +1,1565 @@
+            and number_pending_instances > 0
+        ):
+
+            pending_entries = self.get_oldest_pending_entries(
+                MAX_RUNNING_INSTANCES - number_instances
+            )
+
+            logger.debug(
+                "wake_instance: %i %i"
+                % (number_instances, number_pending_instances)
+            )
+
+            for entry in pending_entries:
+                created_at = entry[0]
+                pid = entry[1]
+                # Change status to not pending and continue the process
+                self.update_pending_status(created_at, False)
+                self.start_process(pid)
+
+    def start_process(self, pid):
+        """Continue a stopped process
+
+        Send a continue signal to the process with given pid
+
+        Arguments:
+            pid {int} -- Process Identification Number
+        """
+        logger.debug("Continue pid: %i" % (pid))
+        os.kill(pid, signal.SIGCONT)
+
+    def stop_process(self, pid):
+        """Stop a running process
+
+        Send a stop signal to the process with given pid
+
+        Arguments:
+            pid {int} -- Process Identification Number
+        """
+        os.kill(pid, signal.SIGSTOP)
+
+    def check_pid(self, pid):
+        """Check for the existence of a process.
+
+        Arguments:
+            pid {int} -- Process Identification Number
+        """
+        try:
+            os.kill(pid, 0)
+        except OSError:
+            return False
+        else:
+            return True
+
+
+def ping(gmp, im):
+    """Checks for connectivity
+
+    This function sends the get_version command and checks whether the status
+    is ok or not.
+    """
+    version = gmp.get_version()
+    status = version.xpath("@status")
+
+    if "200" in status:
+        end_session(im, "GMP OK: Ping successful", NAGIOS_OK)
+    else:
+        end_session(im, "GMP CRITICAL: Machine dead?", NAGIOS_CRITICAL)
+
+
+def status(gmp, im, version, script_args):
+    """Returns the current status of a host
+
+    This functions return the current state of a host.
+    Either directly over the asset management or within a task.
+
+    For a task you can explicitly ask for the trend.
+    Otherwise the last report of the task will be filtered.
+
+    In the asset management the report id in the details is taken
+    as report for the filter.
+    If the asset information contains any vulnerabilities, then will the
+    report be filtered too. With additional parameters it is possible to add
+    more information about the vulnerabilities.
+
+    * DFN-Certs
+    * Logs
+    * Autofp
+    * Scanend
+    * Overrides
+    """
+    params_used = "task=%s autofp=%i overrides=%i apply_overrides=%i" % (
+        script_args.task,
+        script_args.autofp,
+        int(script_args.overrides),
+        int(script_args.apply_overrides),
+    )
+
+    if script_args.use_asset_management:
+        report = gmp.get_reports(
+            type="assets",
```

Comment:
As far as i know we removed the assets report type in GOS 5. Not sure if we should support this part of the code.